src/app/core/auth/auth.interceptor.ts

```diff
@@ -1,44 +1,17 @@
-import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Router } from '@angular/router';
 
-import { throwError, Observable } from 'rxjs';
-import { catchError } from 'rxjs/operators';
-import { SessionService } from './session.service';
+import { Interceptor } from './interceptor';
 
 /**
- * HTTP Interceptor that will interpret authentication-related HTTP calls
+ * HTTP Interceptor that will interpret authentication related HTTP calls
  */
 @Injectable()
-export class AuthInterceptor implements HttpInterceptor {
-	constructor(private router: Router, private sessionService: SessionService) {
-		// Nothing here
-	}
-
-	intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-		return next.handle(req).pipe(
-			catchError(err => {
-				// Grab all the useful stuff out of the error response
-				const status = err?.status ?? 200;
-				const type = err?.error?.type ?? '';
-				const url = err?.url ?? '';
-				const message = err?.error?.message ?? '';
-
-				const stateObject = { status, type, message, url };
-
-				// Go to signin if the user isn't logged in and wasn't already on the signin page
-				if (401 === status && !url.endsWith('auth/signin')) {
-					this.router.navigate(['/signin']);
-				} else if (403 === status) {
-					if ('eua' === type) {
-						this.router.navigate(['/eua']);
-					} else {
-						this.router.navigate(['/access'], { state: stateObject });
-					}
-				}
-
-				return throwError(err);
-			})
-		);
+export class AuthInterceptor extends Interceptor {
+	handleError(err: any): void {
+		const { status, type, message, url } = this.parseError(err);
+		if (status === 403) {
+			this.router.navigate(['/access'], { state: { status, type, message, url } });
+			this.bypassRemainingInterceptors(err);
```

Comment:
This call to the bypass doesn't do anything as the AuthInterceptor is the last one in the chain, but I figured to leave it in a) for standardization and b) as a safeguard in case the order ever changes in the future.